rsc-server/src/org/openrsc/net/ConnectionHandler.java

```diff
@@ -0,0 +1,125 @@
+package org.openrsc.net;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.channel.ChannelStateEvent;
+import org.jboss.netty.channel.ExceptionEvent;
+import org.jboss.netty.channel.MessageEvent;
+import org.jboss.netty.channel.SimpleChannelHandler;
+import org.openrsc.Config;
+import org.openrsc.model.PlayerManager;
+import org.openrsc.model.net.GameLoginHandler;
+import org.openrsc.model.player.Player;
+import org.openrsc.net.packet.Packet;
+import org.openrsc.net.packet.PacketHandler;
+import org.openrsc.net.packet.PacketManager;
+import org.openrsc.util.GameUtils;
+
+/**
+ * The <code>ConnectionHandler</code> handles incoming packet data.
+ */
+public class ConnectionHandler extends SimpleChannelHandler {
+    
+	private ArrayList<String> connections = new ArrayList<String>();
+
+	/**
+	 * The login decoder instance.
+	 */
+	private final GameLoginHandler loginDecoder;
+
+	public ConnectionHandler() {
+		this.loginDecoder = new GameLoginHandler();
+	}
+
+	@Override
+	public void messageReceived(ChannelHandlerContext ctx, MessageEvent event) {
+		Packet packet = (Packet) event.getMessage();
+		if (packet == null) {
+			Logger.getLogger(getClass().getName()).log(Level.WARNING, "Received null packet from client.");
+			return;
+		}
+		final int opcode = packet.getOpcode();
+
+		// Ping Packet
+		if (opcode == 0) {
```

Comment:
I can add 3 constants, for opcodes 0 through 2, but the rest wont need to be defined because if you check the PacketManager class each packet is constructed and added into an array with it's relevant opcode in the same line of code.

Reply:
Just fyi. 0 though 2 are the only hard-coded cases, and they're like this because they're used for the pre-login protocol.
0 is for for the rsa public key stream
1 is for ping
2 is for the client login request, after 2 is accepted the PacketManager will execute all packets for the session